### PR TITLE
Introduce runAcPowerRoutine

### PR DESCRIPTION
**Pull request by 1c3t3a**
Introduce new `dpsl.diagnostics.power.runAcPowerRoutine` API
function to run the AC Power routine.

### CODE REVIEW

File: src/README.md [hunk 1]
```diff
@@ -179,6 +179,12 @@ class Routine {
 | voltageMinDesign | number | Desired minimum output voltage |
 | voltageNow | number | Battery's voltage (V) |
 
+### AcPowerRoutineParams
+| Property Name | Type | Description |
+------------ | ------- | ----------- |
+| expectedStatus | string | The expected status of the AC ('connected', 'disconnected' or 'unknown') |
+| expectedPowerType | string | (Optional) If specified, this must match the type of power supply for the routine to succeed. |
```
Comment (MahmoudAGawad) [inline src/README.md:186]: To be consistent, let's use the same notion as in "RoutineStatus". That is, follow "expectedPowerType" with an * --> "expectedPowerType*" and remove "(Optional)"

File: src/diagnostics_manager.js [hunk 2]
```diff
@@ -96,6 +96,28 @@ class Routine {
...
+   * Runs AC Power test.
+   * @param {!dpsl.AcPowerRoutineParams} params
+   * @return { !Promise<!Routine> }
+   * @public
+   */
+  async runAcPowerRoutine(params) {
+    const functionName = 'runAcPowerRoutine';
+    if (!isSupported(functionName)) {
+      throw new MethodNotFoundError(API_NAME, functionName,
+          /* chromeVersion */ 100);
```
Comment (MahmoudAGawad) [inline src/diagnostics_manager.js:113]: chromeVersion = 105

File: src/types.js [hunk 3]
```diff
@@ -203,6 +203,15 @@ dpsl.AvailableRoutinesList;
  */
 dpsl.RoutineStatus;
 
+/**
+ * Params object of dpsl.diagnostics.power.runAcPowerRoutine()
+ * @typedef {{
+ *   expectedStatus: !string,
+ *   expectedPowerType?: string
```
Comment (MahmoudAGawad) [inline src/types.js:210]: !string

Comment (1c3t3a) [reply src/types.js:210]: Are you sure? I am not an expert with this notion, but dpsl.RoutineStatus also denotes the optional `output?: string` and not `!string`.

Comment (MahmoudAGawad) [reply src/types.js:210]: The way I see it, expectedPowerType? it either exists or not (optional). However, !string is saying that the value must not be null.

I couldn't find supporting resources, so feel free to drop this comment.

**Pull request merged**